### PR TITLE
Stop showing header links when session status is loading

### LINKED ISSUE #90: Stop showing header links when session status is loading
- To stop showing wrong header links before confirming session with `useSession`, use `status` of `session`

https://github.com/van-squad/map-t3/assets/65790344/f9e17144-b661-44ff-b5d2-a23a6bbb0e2f

### PR DESCRIPTION
**Pull request by keento0809**
### Feature Description

- Added `status` of `session` in Header component so that links in Header are not shown until `session` is confirmed.
- Cleaned up codes of `src/server/auth.ts`

https://github.com/van-squad/map-t3/assets/65790344/6c2e92dc-1479-4216-a57c-78f38b04ab8d

### Related Issue

- close #90 

### Something you want to mention

- Still we remain the issue of toggling background-color (When theme is `dark`, we want to show dark background-color from the beginning)

### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/server/auth.ts [hunk 1]
```diff
@@ -76,33 +70,23 @@ export const authOptions: NextAuthOptions = {
...
  *
  * @see https://next-auth.js.org/configuration/nextjs
  */
-export const getServerAuthSession = (ctx: {
-  req: GetServerSidePropsContext["req"];
-  res: GetServerSidePropsContext["res"];
-}) => {
-  return getServerSession(ctx.req, ctx.res, authOptions);
-};
+export const getServerAuthSession = () => getServerSession(authOptions);
```
Comment (Lada496) [inline src/server/auth.ts]: I just read the document and thought we need to pass `ctx` but is it actually wrong??
https://next-auth.js.org/configuration/nextjs#getserversession

Comment (keento0809) [reply src/server/auth.ts]: @Lada496 Actually when I created a new t3-app with `npx create t3-app@latest` and chose options as app router (& next.js 13), I just noticed that contents in `src/server/auth.ts` were slightly different (like the screenshot). 

Probably, either way works fine, but I just felt like in next.js 13 we don't use `getServerSideProps` so it sounds a bit wired to me, and then I followed the way on the screenshot. But I returned this part of the code to the original one so as not to cause any confusion.

スクリーンショット 2023-10-25 3 24 49

Comment (Lada496) [reply src/server/auth.ts]: Oh I didn't know that! Thanks for the update

**Review by Lada496 — APPROVED** [review]

**Review by SachiGoto — APPROVED** [review]
LGTM!

**Pull request merged**